earth_enterprise/integration_test/specs/state_propagation.spec

```diff
@@ -0,0 +1,787 @@
+
+Clean
+* Clean imagery resource "BlueMarble_CancelResourceDuringBuild"
+* Verify that the state of imagery resource "BlueMarble_CancelResourceDuringBuild" is "Cleaned"
+* Verify that the state of imagery project "StatePropagationTest_CancelResourceDuringBuild" is "Succeeded"
+* Verify that the state of imagery resource "i3SF15meter_CancelResourceDuringBuild" is "Succeeded"
+* Verify that the state of imagery resource "USGSLanSat_CancelResourceDuringBuild" is "Succeeded"
+* Verify that the state of imagery resource "SFHiRes_CancelResourceDuringBuild" is "Succeeded"
+
+Rebuild
+* Build imagery project "StatePropagationTest_CancelResourceDuringBuild"
+* Verify that the state of imagery project "StatePropagationTest_CancelResourceDuringBuild" is "Waiting"
+* Wait for imagery project "StatePropagationTest_CancelResourceDuringBuild" to reach state "Succeeded"
+* Verify that the state of imagery resource "BlueMarble_CancelResourceDuringBuild" is "Succeeded"
+
+## Cancel, Resume, Clean, and Rebuild Project During Packgen
+Tags: cancel, clean, resume, rebuild
+
+Build
+* Create and build default project "StatePropagationTest_CancelProjectLate"
+* Wait for imagery project "StatePropagationTest_CancelProjectLate" to reach state "Queued"
+
+Cancel
+* Cancel imagery project "StatePropagationTest_CancelProjectLate"
+* Verify that the state of images for default project "StatePropagationTest_CancelProjectLate" is "Succeeded"
+* Verify that the state of imagery project "StatePropagationTest_CancelProjectLate" is "Canceled"
+
+Resume
+* Resume imagery project "StatePropagationTest_CancelProjectLate"
+* Verify that the state of images for default project "StatePropagationTest_CancelProjectLate" is "Succeeded"
+* Verify that the state of imagery project "StatePropagationTest_CancelProjectLate" is in
+  | State      |
+  |------------|
+  | Waiting    |
+  | InProgress |
+  | Queued     |
+* Wait for imagery project "StatePropagationTest_CancelProjectLate" to reach state "Succeeded"
+* Verify that the state of images for default project "StatePropagationTest_CancelProjectLate" is "Succeeded"
+
+Clean
+* Clean imagery project "StatePropagationTest_CancelProjectLate"
+* Verify that the state of imagery project "StatePropagationTest_CancelProjectLate" is "Cleaned"
+* Verify that the state of images for default project "StatePropagationTest_CancelProjectLate" is "Succeeded"
+
+Rebuild
+* Build imagery project "StatePropagationTest_CancelProjectLate"
+* Wait for imagery project "StatePropagationTest_CancelProjectLate" to reach state "Succeeded"
+* Verify that the state of images for default project "StatePropagationTest_CancelProjectLate" is "Succeeded"
+
+## Clean After Cancel
+Tags: cancel, clean, rebuild
+
+Build
+* Create and build default project "StatePropagationTest_CleanAfterCancel"
+
+Cancel Resource
+* Cancel imagery resource "BlueMarble_CleanAfterCancel"
```

Comment:
Minor: Those resource names could have used the same prefix as the rest of the resources.

Reply:
I probably should have done that to begin with, but at least they're all under the same parent directory. At this point I don't think it's worth the effort.